salt/modules/solarisips.py

```diff
@@ -298,39 +299,81 @@ def version(*names, **kwargs):
 
     '''
-    if names:
-        cmd = ['/bin/pkg', 'list', '-Hv']
-        cmd.extend(names)
-        lines = __salt__['cmd.run_stdout'](cmd).splitlines()
-        ret = {}
-        for line in lines:
-            ret[_ips_get_pkgname(line)] = _ips_get_pkgversion(line)
-        if ret:
-            return ret
-    return ''
+    if len(names) == 0:
+        return ''
+
+    cmd = ['/bin/pkg', 'list', '-Hv']
+    cmd.extend(names)
+    lines = __salt__['cmd.run_stdout'](cmd, ignore_retcode=True).splitlines()
+    ret = {}
+    for line in lines:
+        ret[_ips_get_pkgname(line)] = _ips_get_pkgversion(line)
+
+    # Append package names which are not installed/found
+    unmatched = list(filter(lambda name: not reduce(lambda x, y : x or name in y, ret.keys(), False), names))
+    ret.update(zip(unmatched, [ '' ] * len(unmatched)))
 
+    # Return a string if only one package name passed
+    if len(names) == 1:
+        return list(ret.values())[0]
 
-def latest_version(name, **kwargs):
+    return ret
+
+
+def latest_version(*names, **kwargs):
     '''
-    The available version of the package in the repository.
-    In case of multiple matches, it returns list of all matched packages.
-    Accepts full or partial FMRI.
+    The available version of packages in the repository.
+    Accepts full or partial FMRI. Partial FMRI is returned if the full FMRI
+    could not be resolved.
+
+    If the latest version of a given package is already installed, an empty
+    string will be returned for that package.
+
     Please use pkg.latest_version as pkg.available_version is being deprecated.
 
+    .. versionchanged:: Fluorine
+        Argument changed from 'name' to '*names' tu support multiple packages.
+
     CLI Example:
 
     .. code-block:: bash
 
+        salt '*' pkg.latest_version bash
         salt '*' pkg.latest_version pkg://solaris/entire
+        salt '*' pkg.latest_version postfix sendmail
     '''
-    cmd = ['/bin/pkg', 'list', '-Hnv', name]
-    lines = __salt__['cmd.run_stdout'](cmd).splitlines()
+
+    if len(names) == 0:
+        return ''
+
+    cmd = ['/bin/pkg', 'list', '-Hnv']
+    cmd.extend(names)
+    lines = __salt__['cmd.run_stdout'](cmd, ignore_retcode=True).splitlines()
     ret = {}
     for line in lines:
         ret[_ips_get_pkgname(line)] = _ips_get_pkgversion(line)
-    if ret:
-        return ret
-    return ''
+
+    installed = version(*names)
+
+    if len(names) == 1:
+        # Convert back our result in a dict if only one name is passed
+        installed = { list(ret.keys())[0] if len(ret) > 0 else names[0]: installed }
+
+    for name in ret:
+        if not name in installed:
+            continue
+        if ret[name] == installed[name]:
+            ret[name] = ''
+
+    # Append package names which are not found
+    unmatched = list(filter(lambda name: not reduce(lambda x, y : x or name in y, ret.keys(), False), names))
+    ret.update(zip(unmatched, [ '' ] * len(unmatched)))
```

Comment:
This can be optimized using `itertools.cycle()`.

```python
ret.update(zip(unmatched, itertools.cycle(('',))))
```

You will also need to add the following two imports:

```python
import itertools
from salt.ext.six.moves import zip  # pylint: disable=redefined-builtin
```

The second import there ensures that Python 2 uses a `zip` which returns an iterator instead of a list.